src/compute/src/logging/reachability.rs

```diff
@@ -72,7 +76,7 @@ pub fn construct<A: Allocate>(
 
         use timely::dataflow::operators::generic::builder_rc::OperatorBuilder;
 
-        let construct_reachability = |key: Vec<_>, value: Vec<_>| {
+        let construct_reachability_unarranged = || {
```

Comment:
The changes here need some restructuring. At the moment, `construct_reachability` is called exactly once, but with your PR it'll be called twice, which means that on runtime we do twice the amount of work. I guess we should refactor the code to not use closures. The reason why there are closures is only to get the code in a more readable state, but with the option of having two outputs, this benefit doesn't make much sense anymore.

Also, it's a misnomer because `construct_reachability_unarranged` produces an arrangement, although it's not an arrangement in the canonical format but hashed differently.